AppServer_Java/src/com/google/appengine/api/taskqueue/dev/DevPullQueue.java

```diff
@@ -77,8 +77,22 @@
         return addResponse;
     }
 
+    synchronized TaskQueuePb.TaskQueueDeleteResponse deleteTask(TaskQueuePb.TaskQueueDeleteRequest deleteRequest) {
+        String queueName = deleteRequest.getQueueName();
+        if (!queueName.equals(getQueueName())) {
+            throw new ApiProxy.ApplicationException(
+                    TaskQueuePb.TaskQueueServiceError.ErrorCode.INVALID_REQUEST.getValue());
+        }
+
+        logger.log(Level.FINE, "PullQueue: Sending DeleteRequest to TaskQueue server for " + queueName);
+        TaskQueuePb.TaskQueueDeleteResponse deleteResponse = client.delete(deleteRequest);
+
+        return deleteResponse;
+    }
+
+    // Abstract method needs to be overriden here.
     boolean deleteTask(String taskName) {
-        return this.taskMap.remove(taskName) != null;
+        throw new ApiProxy.ApplicationException(TaskQueuePb.TaskQueueServiceError.ErrorCode.INTERNAL_ERROR.getValue());
```

Comment:
That is not used in the Pull Queues but since both DevPullQueue and DevPushQueue inherit from DevQueue it must be overridden here.

Unfortunately, there is no NOT_IMPLEMENTED error code in the TaskQueuePb class.
I can add a message. Thanks.